extension/src/popup/SorobanContext.tsx

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import * as SorobanClient from "soroban-client";
+
+import { SOROBAN_RPC_URLS } from "@shared/constants/stellar";
+
+import { settingsNetworkDetailsSelector } from "./ducks/settings";
+
+export const SorobanContext = React.createContext({} as SorobanTxBuilder);
+
+export class SorobanTxBuilder {
+  public source: SorobanClient.Account;
+  private fee: string;
+  private pubKey: string;
+  private networkPassphrase: string;
+  public server: SorobanClient.Server;
+
+  constructor(pubKey: string, fee: string, networkPassphrase: string) {
```

Comment:
this type of structure makes sense to me if he were to have multiple instances of this class with different public keys, fees, and network phrases.

Since the inputs and outputs will always be the same (we're only ever dealing with one public key, for ex), does this make sense for this be a function rather than a class?

I also bring it up because we just don't tend to use classes in this codebase, so there's also kind of a stylistic component to this comment, as well.

Reply:
yeah I think that makes sense, I thought it would do more when I first wrote it as a class but I see no reason why it can't be a function or two, will refactor.

Reply:
refactored this in commit 16200e44ab98a1c820f7e795091a04b53b76c325